### ISSUE TITLE
Hangs on splash screen from first run

### ISSUE THREAD

**Issue by ghost** [TTTDevs/NextCloudBookmarksAndroid#4]
When I try to share bookmarks to Nextcloud, I only get to the screen saying "Connecting to NextCloud. Please wait." Nothing more happens, not even a blank screen like in issue #2.

* NextCloud URL is correctly filled in
* Bookmarks active on server instance
* Nothing registers on server side, either.

**Comment by mdev88** [comment]
By any chance, does your server use a self-signed certificate?

**Comment by ghost** [comment]
It does not, no.

**Comment by rNoz** [comment]
I confirm the same behaviour.

In my case it says just: "SSL Certificate Error. Do you want to continue anyway?", It is not detailed what happened. After that, if I say "Continue"  (or "Cancel") it just hangs with the blue screen and the logo saying "Contacting Nextcloud, please wait..."

I don't use self-signed certificates in the server.

**Comment by mdev88** [comment]
@rNoz your problem is the one in #2 related to the self-signed certificate, don't mind the title of the issue.

@haverholm Please try the Test APK I just published and let me know if you see any change.

**Comment by ghost** [comment]
No chance, sorry.

**Comment by alecbl** [comment]
I have the same issue as haverholm. Tested said apk but it did not help. And no, not self-signed cert (Let's Encrypt).

**Comment by Seferi** [comment]
I have the same problem, nextcloudbox, lets encrypt certificate.
Android 7.1.2 LineageOS 14.1

**Comment by Milliw** [comment]
Same for me here.
NextCloud 11.0.3 with https on all-inkl.com server with Let´s encrypt certificate.
Sony Xperia Z3 Compact D5803 with stock Android 6.0.1 and AFWall+ (disabled for testing).

I was wondering if the "bookmarks" app has to be installed in NextCloud cause theres no mention here.

**Comment by mdev88** [comment]
@MarcoWilli yes, you need to enable the Bookmarks app in your server. i mention that in the README, but I should also add that in the app too.

**Comment by 9662** [comment]
Sorry for the me too but… me too!

Android 7.1.1, Let's Encrypt certificate. Nextcloud 11.0.3, bookmarks app installed. Application version 1.1 from F-Droid.

**Comment by c33s** [comment]
same here. 
nexcloud 11.0.3, letsencrypt, app version 1.1 fdroid. android 6.0.1

**Comment by andreas-h** [comment]
Since upgrade to Nextcloud 12, I'm also affected by this bug ...

**Comment by alcos0** [comment]
Same here. I am running CopperheadOS.
adb logacat displays this error:
`I/chromium: [INFO:CONSOLE(1)] "Uncaught SyntaxError: missing ) after argument list", source:  (1)`

**Comment by allan-null** [comment]
I wanted really bad to get this app working, but I am no programmer. I wanted so bad this app I spent the whole night messing with the code to finally manage to replace the JavaScript loading in a webview for a simple URL with variables encoded with a native Java function. Only to then realize once Android kills the webview it also kills the Nextcloud session (Nextcloud default behavior is to set cookies to destroy when the browser closes). I addressed on issue #3 that webview is not the way to go in this case.

**Comment by Gikto** [comment]
Same issue on OnePlus3 Android 7.1.1 and Nextcloud 12

**Comment by e-alfred** [comment]
Same here, Nextcloud 12, Bookmarks App 0.10, Android 7.1.1 and a SSL cert from a major CA (no Letsencrypt).

I get this in the logs on the server:


```
{"reqId":"WT-BzH8AAQEAAGwxAD8AAAAG","level":0,"time":"2017-06-13T12:43:25+02:00","remoteAddr":"1.1.1.1","user":"--","app":"no app in context","method":"GET","url":"\/owncloud\/index.php\/apps\/bookmarks\/bookmarklet?output=popup&url=Save%20to%20Nextcloud%20Bookmarks%20(Save%20bookmarks%20on%20a%20NextCloud%20server)%20-%20https%3A%2F%2Ff-droid.org%2Fapp%2Fcom.tttdevs.stncbookmarks&title=","message":"Current user is not logged in","userAgent":"Mozilla\/5.0 (Linux; Android 7.1.1; Phone Build\/NMF26F; wv) AppleWebKit\/537.36 (KHTML, like Gecko) Version\/4.0 Chrome\/58.0.3029.83 Mobile Safari\/537.36","version":"12.0.0.29"}
{"reqId":"WT-BzX8AAQEAAGwxAEAAAAAG","level":0,"time":"2017-06-13T12:43:26+02:00","remoteAddr":"1.1.1.1","user":"--","app":"core","method":"GET","url":"\/owncloud\/login?redirect_url=\/owncloud\/index.php\/apps\/bookmarks\/bookmarklet%3Foutput%3Dpopup%26url%3DSave%2520to%2520Nextcloud%2520Bookmarks%2520%28Save%2520bookmarks%2520on%2520a%2520NextCloud%2520server%29%2520-%2520https%253A%252F%252Ff-droid.org%252Fapp%252Fcom.tttdevs.stncbookmarks%26title%3D","message":"Scss is disabled for \/var\/www\/owncloud\/core\/css\/server.scss, ignoring","userAgent":"Mozilla\/5.0 (Linux; Android 7.1.1; Phone Build\/NMF26F; wv) AppleWebKit\/537.36 (KHTML, like Gecko) Version\/4.0 Chrome\/58.0.3029.83 Mobile Safari\/537.36","version":"12.0.0.29"}
{"reqId":"WT-BzX8AAQEAAGwxAEAAAAAG","level":0,"time":"2017-06-13T12:43:26+02:00","remoteAddr":"1.1.1.1","user":"--","app":"core","method":"GET","url":"\/owncloud\/login?redirect_url=\/owncloud\/index.php\/apps\/bookmarks\/bookmarklet%3Foutput%3Dpopup%26url%3DSave%2520to%2520Nextcloud%2520Bookmarks%2520%28Save%2520bookmarks%2520on%2520a%2520NextCloud%2520server%29%2520-%2520https%253A%252F%252Ff-droid.org%252Fapp%252Fcom.tttdevs.stncbookmarks%26title%3D","message":"Scss is disabled for \/var\/www\/owncloud\/core\/css\/share.scss, ignoring","userAgent":"Mozilla\/5.0 (Linux; Android 7.1.1; Phone Build\/NMF26F; wv) AppleWebKit\/537.36 (KHTML, like Gecko) Version\/4.0 Chrome\/58.0.3029.83 Mobile Safari\/537.36","version":"12.0.0.29"}

```

**Comment by peturbg** [comment]
hi guys. to the developer: your program was great but in version 1.0, in version 1.1 with nextcloud it dont works, my server is debin 9 the latest stable version all up-to date and no it is not the test of stretch, it is the latest stable debian with http2 supprt. now man, i dont know what you had changed in safe bookmark in nextcloud but it dont work a and i see that when i connect from your program with version 1.0 iy comment me that safe book to nextcloud is chrome. but why chrome, why you just dont use firefox it is more secure and dont leaks private data. man thanks again for your program, that works on version 1.0 and it didnt works in version 1.1 and man when you update bookmanager for nextcloud, put more things, jpeg download, becouse i had bookmark and now i dont know what site is what. man make folders with book marks, for example folder for download iso. and i use ssl certificate legal. thanks

**Comment by e-alfred** [comment]
Indeed, going back to version 1.0 from F-Droid and it works. This seems to be a major regression in the 1.1 version of the app.

**Comment by mdev88** [comment]
Version 1.1 wasn't supposed to be posted on F-droid ¬¬
I thought they were picking up the app from the "releases" (tags). Version 1.1 is a testing version for trying to debug these issues.
I just cannot reproduce them. Can anyone give me a Nextcloud's url so that I can try to see the error myself?

**Comment by ghost** [comment]
Right, so I took the recent advice and downgraded to v.1.0 from F-droid. Now I get a "Web page not available" error trying to save the first bookmark:

>The web page at http://my.nextcloud.com/index.php/apps/bookmarks/bookmarklet?output=popup&url=http%3A%2F%2Fshared_url.com%2F&title= might be temporarily down or it may have moved permanently to a new web address.

**Comment by radio070201** [comment]
Well yesterday I had same problem. CM13 and. Nextcloud 12 here, downloaded v1.1 from F-DROID and just hanging. After reading this issue here in github I unistalled v1.1 and installed v1.0 also from F-DROID. The problem is solved and app is working fine. 
So v1.1 is broken but v1.0 is ok. 

Just to be clear I DO suggest to write on app that the expected URL is the server only, its main URL only. And not the url of the bookmarks app (....server/index.php/bookmarks) 

Thanks for nice app! I hope it evolves to something like Basket, Pocket. 
By the way, how can we access/read the bookmarks on cloud on Android? And on Windows? Only way now is to open server web interface on a browser? 
I also hope someday we have something like Xmarks for nextcloud/OwnCloud.

**Comment by e-alfred** [comment]
>By the way, how can we access/read the bookmarks on cloud on Android?

There is this Open Source Android app:

https://github.com/lenchan139/NCBookmark

**Comment by radio070201** [comment]
> There is this Open Source Android app:
https://github.com/lenchan139/NCBookmark

Thanks! Tried it now but unfortunately it is almost useless yet: does not support bookmarks with more than 1 tag (and other bugs). I'll keep an eye on it. 
Maybe this app here can integrate some of its code so that we have only 1 app to write and read nextcloud bookmarks. 

Thanks.

**Comment by peturbg** [comment]
hi github silence my profile becouse someone lie that telling the true veraion 1.1 dont work and version 1.0 work. 2. the developer tttdevelopers comment version 1.1 dont has to be in f-droid so from martinchodev
commented 24 days ago

Version 1.1 wasn't supposed to be posted on F-droid ¬¬
I thought they were picking up the app from the "releases" (tags). Version 1.1 is a testing version for trying to debug these issues.
I just cannot reproduce them. Can anyone give me a Nextcloud's url so that I can try to see the error myself?
now developer lie. 3rd from 24 days and version 1.1 continued be in f-droid and developers didnt remove it. so she is stupid. orthe developer wants to do bas of the people or she has serious problems. removing a program can be inmediatelly ir in 7 works days. we live in 21 centuey where you contact instantelly and write email is inmediatelly max. it can take 7 day to remove a shit program. but no version 1.1 is still active from 24 days and developers disnt do new versions. developers had read my messages that they program version 1.1 dont work yy can see my emails upper and peoplea too confurming version 1.0 work. developers can patch and make patch inmediatelly. we dont leave in stone age, if developera delete they source code of the program they can make it again include github is for source so they didnt delete they source. 4th bookmanager client for android use obsolete version of google chrome, so developers if you are gonna do book manager for android forat make it with firefox and autoupdate. i dont care the programs have updates to fix holes. include i dont see peoples complaining to developers make news version or fix version 1.1. this mean you can give free shit of human and they will eat it. shhhh dont intent complain, if you dont complain for fix version 1.1 then you cant complaint to eat shit.

**Comment by creideiki** [comment]
I'm seeing the same issue ("Connecting to NextCloud. Please wait." forever) with version 1.1 from F-Droid. Dowgrading to 1.0 fixed it.

- Nextcloud 12.0.0
- Certificate signed by a CA I have manually added to Android
- Fairphone 2 running Fairphone Open 17.06.4, which is based on Android 6.0.1

Web server logs say the v1.1 app retrieves the login page, and then does nothing:
```
$IP - - [16/Jul/2017:01:20:02 +0200] "GET /nextcloud/index.php/apps/bookmarks/bookmarklet?output=popup&url=xxx HTTP/1.1" 303 -
$IP - - [16/Jul/2017:01:20:02 +0200] "GET /nextcloud/index.php/login?redirect_url=/nextcloud/index.php/apps/bookmarks/bookmarklet%3Foutput%3Dpopup%26url%3Dxxx%26title%3Dyyy HTTP/1.1" 200 8074
```
